File: 785.py
a, b = map(int, input().split())

def is_automorph(x: int) -> bool:
    sq = x * x
    while x > 0:
        if x % 10 != sq % 10:
            return False
        x, sq = x // 10, sq // 10
    return True

print(" ".join(
    str(x) for x in range(a, b+1) if is_automorph(x)
))

File: 785.cpp
#include <iostream>

bool is_automorph(long long x) {
    for (long long sq = x * x; x > 0; x /= 10, sq /= 10) {
        if (x % 10 != sq % 10) return false;
    }
    return true;
}

int main() {
    int a, b; std::cin >> a >> b;
    for (int i = a; i <= b; ++i) {
        if (is_automorph(i)) {
            std::cout << i << ' ';
        }
    }
}